content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -1,72 +1,48 @@
 2. The `accept()` function of the ServerSocket class is called by the server. This method waits for a client to connect to the server on a specific port before returning.
 3. A client builds a `Socket` object, giving the name of the server and the connecting port number as the server awaits.
 4. The `Socket` class constructor will try to connect the client to the server and port number provided. The client now has a `socket` object that if communication is established, may connect with the server.
 5. The `accept()` method on the server produces a reference to a new server socket that is connected to the client's socket.
  
-If the connection is successful, client and server can then communicate using **I/O streams**. The **I/O streams** come from socket objects of the client and server. The client's `OutputStream` connects to the server's `InputStream` while the server's `OutputStream` connects to the client's `InputStream`.
-
-Incase you don't know, a stream is a collection sequenced data. They can be divided into two categories:
+If the connection is successful, the client and server can then communicate using **I/O streams**. The **I/O streams** come from socket objects of the client and server. The client's `OutputStream` connects to the server's `InputStream` while the server's `OutputStream` connects to the client's `InputStream`.
+In case you don't know, a stream is a collection of sequenced data. They can be divided into two categories:
 1. Character stream (usually used with text files)
 2. Byte stream (usually used with images)
-
 Character stream is in human-readable language while byte stream is in machine language, a language understandable by the CPU.
-
-In our case, the client and server will send messages to each other's streams in parallel.This is made possible because TCP is a two-way communication protocol.
+In our case, the client and server will send messages to each other's streams in parallel. This is made possible because TCP is a two-way communication protocol.
    
 ### Server-side Programming 
-
-A server socket is an instance of `ServerSocket` class and can be created by one of the following constructor: 
-
+A server socket is an instance of the `ServerSocket` class and can be created by one of the following constructors: 
 ```java
 serversocket = new Serversocket (int port)
 ```
-
 port: port number at which the server will be listening for requests from clients.
-
 The ServerSocket object sits and waits for network requests to arrive. Our client will try to connect to port number 5000 that we selected. 
-
-**Note** Make sure the ports are same otherwise there will be no connection
+**Note** Make sure the ports are the same otherwise there will be no connection
 Two while loops will be in use: 
-
 **First Loop**- This is to ensure the server is constantly running.
```

Comment:
> **First Loop**- This is to ensure the server is constantly running.
> **Second Loop**- Ensures that once the client is connected, the server interacts with the client until the client disconnects.

I think I mentioned this before but I would change this to:

> **First Loop** is to ensure the server is constantly running.
> **Second Loop** is to ensure that once the client is connected, the server interacts with the client until the client disconnects.

To fix the formatting and flow.

Reply:
I may have said something different last time but this would also work.